app/gateway/2.8.x/admin-api/developers/reference.md

```diff
@@ -0,0 +1,1439 @@
+---------:                  | --------   
+`{ROLE_NAME|ROLE_ID}` <br>*required*       | The name or UUID of the role you are inspecting.
+
+**Response**
+
+```
+HTTP/1.1 204 No Content
+```
+
+
+## Applications
+
+### List all applications for a developer
+
+**Endpoint**
+
+<div class="endpoint get">/developers/{DEVELOPER_EMAIL|DEVELOPER_ID}/applications</div>
+
+Attribute                         | Description
+---------:                        | --------   
+`{DEVELOPER_EMAIL|DEVELOPER_ID}`  | The email or UUID of a developer.
+
+
+**Response**
+```
+HTTP/1.1 200 OK
+```
+
+```json
+{
+    "data": [
+        {
+            "consumer": {
+                "id": "6e31bf1e-dbcb-4a31-bac9-a192fa24f088"
+            },
+            "created_at": 1644963627,
+            "developer": {
+                "id": "5f60930a-ad12-4303-ac5a-59d121ad4942"
+            },
+            "id": "5ff48aaf-3951-4c99-a636-3b682081705c",
+            "name": "example_app",
+            "redirect_uri": "http://mockbin.org",
+            "updated_at": 1644963627
+        },
+        {
+            "consumer": {
+                "id": "2390fd02-bbcd-48f1-b32f-89c262fa68a8"
+            },
+            "created_at": 1644963657,
+            "developer": {
+                "id": "5f60930a-ad12-4303-ac5a-59d121ad4942"
+            },
+            "id": "94e0f633-e8fd-4647-a0cd-4c3015ff2722",
+            "name": "example_app2",
+            "redirect_uri": "http://mockbin.org",
+            "updated_at": 1644963657
+        }
+    ],
+    "next": null,
+    "total": 2
+}
+
+```
+
+### Inspect an application
+
+**Endpoint**
+
+<div class="endpoint get">/developers/{DEVELOPER_EMAIL|DEVELOPER_ID}/applications/{APPLICATION_ID}</div>
+
+Attribute                         | Description
+---------:                        | --------   
+`{DEVELOPER_EMAIL|DEVELOPER_ID}`  | The email or UUID of a developer.
+`{APPLICATION_ID}`  | The application UUID.
+
+**Response**
+```
+HTTP/1.1 200 OK
+```
+
+```json
+{
+    "consumer": {
+        "id": "d5ea9628-f359-4b96-9f76-61562aaf4300"
+    },
+    "created_at": 1644965555,
+    "custom_id": "billing",
+    "developer": {
+        "id": "5f60930a-ad12-4303-ac5a-59d121ad4942"
+    },
+    "id": "ca0d62bd-4616-4b87-b947-43e33e5418f0",
+    "name": "testapp",
+    "redirect_uri": "http://mockbin.org",
+    "updated_at": 1644965555
+}
+```
+
+
+### Create an application
+
+**Endpoint**
+
+<div class="endpoint post">/developers/{DEVELOPER_EMAIL|DEVELOPER_ID}/applications</div>
+
+
+Attribute                         | Description
+---------:                        | --------   
+`{DEVELOPER_EMAIL|DEVELOPER_ID}`  | The email or UUID of a developer.
+
+**Request body**
+
+Attribute                    | Description
+---------:                   | --------   
+`name`<br>*required*         | A name for the application.
+`redirect_uri`<br>*required* | The application's URI.
+`custom_id`                  | A custom description for the application.
```

Comment:
What is the purpose of the `custom_id`?

Reply:
The `custom_id` will be saved to the linked `consumer` entity for the application. This can be used for OIDC claim mapping when configuring the application's `openid-connect` authentication plugin.